mPower2/mPower2/View Controllers/TaskBrowserViewController.swift

```diff
@@ -0,0 +1,366 @@
+            rule.rsd_alignToSuperview([.leading, .trailing], padding: 0.0)
+            rule.rsd_makeHeight(.equal, 1.0)
+        }
+        
+        // Finally, add a shadow if required
+        if shouldShowTopShadow {
+            let shadow = RSDShadowGradient()
+            shadow.translatesAutoresizingMaskIntoConstraints = false
+            view.addSubview(shadow)
+            shadow.rsd_makeHeight(.equal, 5.0)
+            shadow.rsd_alignToSuperview([.leading, .trailing], padding: 0.0)
+            shadow.rsd_alignToSuperview([.top], padding: -5.0)
+        }
+    }
+    
+    // MARK: RSDTaskViewControllerDelegate
+    func taskController(_ taskController: RSDTaskController, didFinishWith reason: RSDTaskFinishReason, error: Error?) {
+        // dismiss the view controller
+        (taskController as? UIViewController)?.dismiss(animated: true) {
+        }
+        
+        print("\n\n=== Completed: \(reason) error:\(String(describing: error))")
+        print(taskController.taskPath.result)
+    }
+    
+    func taskController(_ taskController: RSDTaskController, readyToSave taskPath: RSDTaskPath) {
+        //
+    }
+    
+    func taskController(_ taskController: RSDTaskController, asyncActionControllerFor configuration: RSDAsyncActionConfiguration) -> RSDAsyncActionController? {
+        return nil
+    }
+
+    // MARK: TaskBrowserTabViewDelegate
+    func taskGroupSelected(identifier: String) {
+        
+        // Save our selected task group and reload collection
+        selectedTaskGroup = taskGroups?.filter({ $0.identifier == identifier }).first
+        collectionView.reloadData()
+        
+        // Now update the isSelected value of all the tabs
+        tabButtonStackView.arrangedSubviews.forEach({
+            if let tabView = $0 as? TaskBrowserTabView {
+                tabView.isSelected = tabView.taskGroupIdentifier == identifier
+            }
+        })
+    }
+}
+
+extension TaskBrowserViewController: UICollectionViewDelegate, UICollectionViewDataSource, UICollectionViewDelegateFlowLayout {
+    
+    // MARK: UICollectionViewDataSource
+    func collectionView(_ collectionView: UICollectionView, numberOfItemsInSection section: Int) -> Int {
+        guard let selectedTaskGroup = selectedTaskGroup else {
+            return 0
+        }
+        return selectedTaskGroup.tasks.count
+    }
+    
+    // MARK: UICollectionViewDelegate
+    func collectionView(_ collectionView: UICollectionView, cellForItemAt indexPath: IndexPath) -> UICollectionViewCell {
+        
+        let cell = collectionView.dequeueReusableCell(withReuseIdentifier: kCollectionCellIdentifier, for: indexPath) as? TaskCollectionViewCell
+        if let selectedTaskGroup = selectedTaskGroup {
+            
+            let task = selectedTaskGroup.tasks[indexPath.row]
+            cell?.image = nil
+            task.imageVendor?.fetchImage(for: CGSize(width: 0.0, height: 0.0)) { (_, img) in
```

Comment:
Please use `.zero` instead of `CGSize(width: 0.0, height: 0.0)` or better yet use the size from the collectionView layout.

Reply:
done